runtime/compiler/trj9/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -3943,7 +3943,9 @@ J9::Z::TreeEvaluator::ArrayStoreCHKEvaluator(TR::Node * node, TR::CodeGenerator
       {
       // If we have an actionable wrtbar and a NULL ptr, branch around the wrt bar
       // as it needs not be exec'd
-      generateS390CompareAndBranchInstruction(cg, TR::InstOpCode::getCmpOpCodeFromNode(sourceChild), node, srcReg, 0, TR::InstOpCode::COND_BE, (doWrtBar || doCrdMrk)?simpleStoreLabel:wbLabel, false, true);
+      TR::InstOpCode::Mnemonic compareAndBranchMnemonic = usingCompressedPointers ? TR::InstOpCode::C : TR::InstOpCode::getCmpOpCode();
+
+      generateS390CompareAndBranchInstruction(cg, compareAndBranchMnemonic, node, srcReg, 0, TR::InstOpCode::COND_BE, (doWrtBar || doCrdMrk)?simpleStoreLabel:wbLabel, false, true);
```

Comment:
For this change we didn't change the semantics of the call at all. We merely pulled up the logic to be inside of this function.